runtime/jcl/common/java_lang_ref_Reference.cpp

```diff
@@ -63,4 +63,39 @@ Java_java_lang_ref_Reference_waitForReferenceProcessingImpl(JNIEnv *env, jclass
 	return result;
 }
 
+#if JAVA_SPEC_VERSION >= 16
+JNIEXPORT jboolean JNICALL
+java_lang_ref_Reference_refersTo(JNIEnv *env, jobject reference, jobject target)
+{
+	J9VMThread * const currentThread = (J9VMThread *)env;
+	J9JavaVM * const vm = currentThread->javaVM;
+	J9InternalVMFunctions * const vmFuncs = vm->internalVMFunctions;
+	jboolean result = JNI_FALSE;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	if (NULL == reference) {
+		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		j9object_t j9reference = J9_JNI_UNWRAP_REFERENCE(reference);
+		j9object_t j9target = (NULL != target) ? J9_JNI_UNWRAP_REFERENCE(target) : NULL;
+		j9object_t referent = NULL;
+
+		if (J9_GC_POLICY_METRONOME == ((OMR_VM *)vm->omrVM)->gcPolicy) {
+			referent = vm->memoryManagerFunctions->j9gc_objaccess_referenceGet(currentThread, j9reference);
+		} else {
+			referent = J9VMJAVALANGREFREFERENCE_REFERENT_VM(vm, j9reference);
+		}
```

Comment:
@RSalman This pattern repeats a few more times in the code. 
1) This needs to be more generic and refer to SATB barrier, rather than Metronome GC policy
2) We should hide this SATB specialization under the hood of J9VMJAVALANGREFREFERENCE_REFERENT_VM macro

Reply:
Those macros are currently autogenerated, so we should add a new pair of macros to get the referent (leaving the autogenerated ones as they are).

Reply:
I would suggest doing this in a separate PR.

Reply:
yeah, I did not mean doing anything now

Reply:
Noted this: https://github.com/eclipse/omr/issues/5494#issuecomment-724780501